### ISSUE TITLE
Limit file types on upload for transcripts

### ISSUE THREAD

**Issue by elynema** [avalonmediasystem/avalon#5523]
# Description

Currently, it appears that users can upload any type of file as a transcript or supplemental file. This should be ok for supplemental files, which are just made available to users to download. However, transcripts will only display properly to users if they are limited to specific file types (captions are already restricted to VTT and SRT files).

Currently, I was able to upload a .mov file as a transcript. It causes the transcripts tab to appear and shows up as an option in the transcript drop-down, but doesn't display content and can't be downloaded.

<img width="1508" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOGxaQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--056748856f5f8bb0dbbf5a6093f50de35e635e6c/image.png" alt="image.png" />

# Done Looks Like
- [ ] Users can only upload .txt, .pdf, .docx, .srt or .vtt files as transcripts
- [x] Add validate_transcript method to run before saves
- [x] If users try to upload an unsupported file type, they should get an error that indicates permitted file types

# Questions
- [ ] Should we provide a tooltip in the interface so users can know what file types are accepted?

**Comment by elynema** [comment]
Not a 7.7 issue.

**Comment by elynema** [comment]
As for restricting upload, we already have the validation on saving a Supplemental File that checks the mime type for text/vtt or text/srt if it is meant to be a caption so it should not be possible for users to upload wrong-typed files through the section file edit form.

**Comment by elynema** [comment]
@joncameron UMD had a use case for wanting to upload .pdf transcript. Have we decided we aren't going to continue to support that use case?

**Comment by joncameron** [comment]
@elynema Definitely want to support PDF, it's been added to the done looks like as a supported format.

**Comment by elynema** [comment]
@joncameron I am wondering how we want the transcript component to work. Do we want it to be open upload so that users can label any type of file a transcript (ex: .doc is not currently on the allowed list) for download, even if it doesn't display? Or do we want users to only be able to upload file types that will actually appear as transcript (.pdf already breaks this)?

**Comment by elynema** [comment]
@joncameron and I discussed. We would prefer to allow any file type to be uploaded as a transcript and downloadable, but only display content in the component if it. @masaball Does that mean this issue is moot?

I thought I saw an example where Aviary does render PDFs with the player component, which is an interesting idea, but I can't find it again at this point, so we'd need to reach out to Kevin Glick to try to learn more.
